src/xcode/ENA/ENA/Source/Scenes/Onboarding/OnboardingInfoViewController.swift

```diff
@@ -227,11 +228,19 @@ final class OnboardingInfoViewController: UIViewController {
 		stateStateLabel.text = exposureNotificationsEnabled ? onboardingInfo.stateActivated : onboardingInfo.stateDeactivated
 
 		guard !pageSetupDone else {
+			// Back button tapped
+			// If "logging of contacts" has been enabled then display alternative text on buttom
+			if pageType == .enableLoggingOfContactsPage && showStateView {
+				nextButton.setTitle(onboardingInfo.alternativeActionText, for: .normal)
+			}
```

Comment:
Seems like you set the alternative title independent of whether `pageSetupDone` is `true` or `false`. I think you could just move this if statement above the guard and remove it from the switch below.

Reply:
And maybe move it to, or combine it with, the already existing line:
"nextButton.setTitle(onboardingInfo.actionText, for: .normal)"

Reply:
Thanks, I've done some refactoring.

Reply:
Separated updateUI() and pageSetup(); setting the alternative button text is now independent of pageSetupDone